Xamarin.Mac: MM5309 error when using Xcode earlier than Xcode 12

### Steps to Reproduce

1. Create a new Xamarin.Mac project
2. Build with Xcode 11

Native linking failure with Xcode 11.3.1

```
Undefined symbols for architecture x86_64:
  "___darwin_check_fd_set_overflow", referenced from:
      _ves_icall_System_ConsoleDriver_InternalKeyAvailable in libmonosgen-2.0.a(libmonoruntimesgen_la-console-unix.o)
      _socket_transport_connect in libmonosgen-2.0.a(libmono_dbg_la-debugger-agent.o)
      _monoeg_g_spawn_command_line_sync in libmonosgen-2.0.a(libeglib_la-gspawn.o)
      _mono_poll in libmonosgen-2.0.a(libmonoutils_la-mono-poll.o)
  "_pthread_jit_write_protect_np", referenced from:
      _free_chunklist in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _new_codechunk in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _mono_codeman_enable_write in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _mono_codeman_disable_write in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _codechunk_valloc in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

error MM5309: Failed to execute the tool 'clang', it failed with an error code '1'. Please check the build log for details.
```


Native linking failure with Xcode 11.7

```
Undefined symbols for architecture x86_64:
  "_pthread_jit_write_protect_np", referenced from:
      _free_chunklist in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _new_codechunk in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _mono_codeman_enable_write in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _mono_codeman_disable_write in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
      _codechunk_valloc in libmonosgen-2.0.a(libmonoutils_la-mono-codeman.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

error MM5309: Failed to execute the tool 'clang', it failed with an error code '1'. Please check the build log for details.


```

--- a/tools/common/Driver.cs
+++ b/tools/common/Driver.cs
@@ -31,7 +31,13 @@ namespace Xamarin.Bundler {
 		public static bool IsUnifiedFullSystemFramework { get { return TargetFramework == TargetFramework.Xamarin_Mac_4_5_System; } }
 		public static bool IsUnifiedMobile { get { return TargetFramework == TargetFramework.Xamarin_Mac_2_0_Mobile; } }
 
+#if MMP
+		// We know that Xamarin.Mac apps won't compile unless the developer is using Xcode 12+: https://github.com/xamarin/xamarin-macios/issues/11937, so just set that as the min Xcode version.
+		static Version min_xcode_version = new Version (12, 0);
+#else
 		static Version min_xcode_version = new Version (6, 0);
+#endif
+
 #if !NET
 		public static int Main (string [] args)
 		{

